08-compiler-features.tex

```diff
@@ -36,6 +139,143 @@ \section{Dead Code Elimination}
 \section{Completion}
 \label{cr-completion}
 
+Because of the high number of Haxe features and its powerful type inference, \href{http://haxe.org/documentation/introduction/editors-and-ides.html}{IDE and code editors} cannot easily handle completion by parsing the Haxe files. Haxe provides completion support for IDE directly built in the compiler thanks to the \ic{--display} commandline parameter.
+
+Let's have a look at the following example :
+\begin{lstlisting}
+class Test {
+    public static function main() {
+        trace("Hello".|
+    }
+}
+\end{lstlisting}
+
+The vertical bar indicates the cursor position after the dot has been pressed. At this time, this is the job of the IDE to call the Haxe compiler with usual compilation parameters, plus an additional one \ic{--display Test.hx@73}.
+
+This parameter tells the file in which we want some completion and the byte position of the cursor in the file. In that example if you count characters with windows CRLF lines ending, you should get 73.
+
+Haxe will perform all the parsing and typing as it would do for normal compilation, except that it will not generate any code. If during its compilation it is reaching the byte 73 in file Test.hx it will output the informations about the type of the expression that is on the left of the dot.
+
+In our case, that would be String. Except that in the cases of an anonymous object or a class instance, the compiler will instead list the different fields in an XML and print them to the standard error output. Here's an example of output:
```

Comment:
\type formatting in "String" missing. Consider review: "The exeption being that..."